src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -43,6 +27,40 @@ public PineconeConnection createIndexIfNotExists(int dimension) throws IOExcepti
                         .withConnectionUrl("https://" + host));
     }
 
+    public String createIndexIfNotExistsControlPlane(PineconeClientConfig config, int dimension) throws IOException, InterruptedException {
+        PineconeIndexOperationClient controlPlaneClient = new PineconeIndexOperationClient(config);
+        List<String> indexList = controlPlaneClient.listIndexes();
+        String indexName = checkIfIndexExists(indexList, dimension, controlPlaneClient);
+
+        return (indexName.isEmpty()) ? createNewIndex(dimension, controlPlaneClient) : indexName;
+    }
+
+    private static String checkIfIndexExists(List<String> indexList, int dimension, PineconeIndexOperationClient controlPlaneClient)
+            throws IOException, InterruptedException {
+        int i = 0;
+        while (i < indexList.size()) {
```

Comment:
Compare to former approach of checking for the first index of the index list only, check for all indexes in the list if there's a potential index that can be used for the test. If not, create a new index. This has helped me cut down total time of the integration tests by 67% (from 10 mins to less than 3 mins).